src/RA_Dlg_AchievementsReporter.cpp

```diff
@@ -172,25 +171,36 @@ INT_PTR CALLBACK Dlg_AchievementsReporter::AchievementsReporterProc(HWND hDlg, U
                     std::string sBugReportComment = ra::Narrow(sBugReportCommentIn);
 
                     //	Intentionally MBCS
-                    char sBugReportInFull[8192];
-                    sprintf_s(sBugReportInFull, 8192,
-                        "--New Bug Report--\n"
-                        "\n"
-                        "Game: %s\n"
-                        "Achievement IDs: %s\n"
-                        "Problem: %s\n"
-                        "Reporter: %s\n"
-                        "ROM Checksum: %s\n"
-                        "\n"
-                        "Comment: %s\n"
-                        "\n"
-                        "Is this OK?",
-                        g_pCurrentGameData->GameTitle().c_str(),
-                        sBuggedIDs,
-                        sProblemTypeNice,
-                        RAUsers::LocalUser().Username().c_str(),
-                        g_sCurrentROMMD5.c_str(),
-                        sBugReportComment.c_str());
+                    std::string sBugReportInFull{ "--New Bug Report--\n\nGame: " };
+                    sBugReportInFull.append(g_pCurrentGameData->GameTitle());
+                    sBugReportInFull.append("\nAchievement IDs: ");
+                    sBugReportInFull.append(sBuggedIDs);
+                    sBugReportInFull.append("\nProblem: ");
+                    sBugReportInFull.append(sProblemTypeNice);
+                    sBugReportInFull.append("\nReporter: ");
+                    sBugReportInFull.append(RAUsers::LocalUser().Username());
+                    sBugReportInFull.append("\nROM Checksum: ");
+                    sBugReportInFull.append(g_sCurrentROMMD5);
+                    sBugReportInFull.append("\n\nComment: ");
+                    sBugReportInFull.append(sBugReportComment);
+                    sBugReportInFull.append("\n\nIs this OK?");
+
+                    if (sBugReportInFull.length() > 8192U)
+                    {
+                        ra::tstring stMsg;
+                        {
+                            std::basic_ostringstream<TCHAR> oss;
+                            oss << _T("Bug report is too long, it needs to be ")
+                                << sBugReportInFull.length() - 8192U
+                                << _T(" characters shorter.\n\n"
+                                    "Consider breaking up this bug report and try again!");
+                            stMsg = oss.str();
+                        }
+                        if (::MessageBox(nullptr, stMsg.c_str(), _T("Error"), MB_RETRYCANCEL | MB_ICONERROR) == IDRETRY)
+                            return TRUE; // User wants to try again
+                        FORWARD_WM_CLOSE(hDlg, ::PostMessage); // User doesn't want to retry
+                        return FALSE;
```

Comment:
I've never actually gotten the bug report to exceed `8192` characters but wanted to see what we could do if that did happen. I just used `100 `instead of `8192`. If it's too long, app will ask if you want to try again and return you to the dialog. If the user doesn't want try again, the dialog is closed going back to the app window. If it succeeded it'll ask you if it was OK like usual, I didn't send the report though.